### PR TITLE
Run server migration steps in correct sequence

### PR DESCRIPTION
**Pull request by keichwa**
https://github.com/SUSE/spacewalk/issues/24788

### CODE REVIEW

**Comment by keichwa** [comment]
Wondering whether we need a `ssh-copy-id` step somewhere?

File: modules/installation-and-upgrade/pages/container-deployment/suma/server-migration-suma.adoc [hunk 1]
```diff
@@ -26,16 +26,18 @@ Self trusted GPG keys are not migrated.
 GPG keys that are trusted in the RPM database only are not migrated.
 Thus synchronizing channels with [command]``spacewalk-repo-sync`` can fail.
 
-The administrator must migrate these keys manually from the 4.3 installation **after migration**:
+The administrator must migrate these keys manually from the 4.3 installation to the container host before the actual server migration.
```
Comment (Bischoff) [inline modules/installation-and-upgrade/pages/container-deployment/suma/server-migration-suma.adoc:29]: I am ignorant about that part (the GPG keys).

**Review by Bischoff — COMMENTED** [review]
No strong feelings.

Besides, I am ignorant with respect to the section about GPG keys. I never tested that. I just know we missed it when we migrated our internal instance.

File: modules/installation-and-upgrade/pages/container-deployment/suma/server-migration-suma.adoc (hunk 1 above)
Comment (keichwa) [reply modules/installation-and-upgrade/pages/container-deployment/suma/server-migration-suma.adoc]: I changed this again. I should have thought twice about it.

File: modules/installation-and-upgrade/pages/container-deployment/suma/server-migration-suma.adoc [hunk 2]
```diff
@@ -59,33 +64,54 @@ systemctl stop postgresql
...
+
+// . This step is optional:
+//   The migration script only uses the 4.3 server's FQDN in the SSH command.
+//   This means that every other configuration required to connect, needs to be defined in the [path]``~/.ssh/config`` file.
+
+. Establish an SSH connection from the new server to the old {productname} Server to check that no password is needed.
+  Also there must not by any problem with the host fingerprint.
+  In case of trouble, remove old fingerprints from the [path]``~/.ssh/known_hosts`` file.
+  Then try again.
+  The fingerprint will be stored in the local [path]``~/.ssh/known_hosts`` file.
```
Comment (Bischoff) [inline modules/installation-and-upgrade/pages/container-deployment/suma/server-migration-suma.adoc:100]: that's perfect!

**Review by Bischoff — APPROVED** [review]

**Comment by keichwa** [comment]
> You cannot copy the new server's key before it is generated. So we have to mention
> 
> ```
> ssh-keygen -t rsa
> ```
> 
> in the process.

Yes, for completeness, and to ease it for all persons involved, I should better start with this.

File: modules/installation-and-upgrade/pages/container-deployment/suma/server-migration-suma.adoc [hunk 3]
```diff
@@ -59,33 +64,62 @@ systemctl stop postgresql
...
+
+. The SSH configuration and agent should be ready on the new server for a passwordless connection to the 4.3 server.
 +
 
 [NOTE]
 ====
 To establish a passwordless connection, the migration script relies on an SSH agent running on the {productnumber} server.
 If the agent is not active yet, initiate it by running [command]``eval $(ssh-agent)``.
-Then, add the SSH key to the running agent with [command]``ssh-add /path/to/the/private/key``.
+Then, add the SSH key to the running agent with [command]``ssh-add``; if the key is not in a standard file, provide it as an argument such as [command]``ssh-add </path/to/the/private/key>``.
```
Comment (Bischoff) [inline modules/installation-and-upgrade/pages/container-deployment/suma/server-migration-suma.adoc:83]: "if the key is not in a standard file" ???

this sounds plain wrong to me -- where does that come from ?

suggested:

Then, add the SSH key to the running agent with [command]``ssh-add`` followed by the path to the private key.

File: modules/installation-and-upgrade/pages/container-deployment/suma/server-migration-suma.adoc [hunk 4]
```diff
@@ -59,33 +64,62 @@ systemctl stop postgresql
...
-. The SSH configuration and agent should be ready on the new {productnumber} server for a passwordless connection to the 4.3 server.
+
+. Ensure that for [systemitem]``root`` an SSH key exists on the new {productnumber} server.
+  If a key does not exist, create it with:
++
+
+----
+ssh-keygen -t rsa
+----
+
```
Comment (Bischoff) [inline modules/installation-and-upgrade/pages/container-deployment/suma/server-migration-suma.adoc:75]: ok

**Review by Bischoff — CHANGES_REQUESTED** [review]
see other change

File: modules/installation-and-upgrade/pages/container-deployment/suma/server-migration-suma.adoc [hunk 5]
```diff
@@ -59,33 +64,62 @@ systemctl stop postgresql
...
+. The SSH configuration and agent should be ready on the new server for a passwordless connection to the 4.3 server.
 +
 
 [NOTE]
 ====
 To establish a passwordless connection, the migration script relies on an SSH agent running on the {productnumber} server.
```
Comment (Bischoff) [inline modules/installation-and-upgrade/pages/container-deployment/suma/server-migration-suma.adoc:81]: on which server?

I guess on the new one

File: modules/installation-and-upgrade/pages/container-deployment/suma/server-migration-suma.adoc (hunk 3 above)
Comment (keichwa) [reply modules/installation-and-upgrade/pages/container-deployment/suma/server-migration-suma.adoc:83]: ok

File: modules/installation-and-upgrade/pages/container-deployment/suma/server-migration-suma.adoc (hunk 5 above)
Comment (keichwa) [reply modules/installation-and-upgrade/pages/container-deployment/suma/server-migration-suma.adoc:81]: yes.  I think I better replace with "new" here.

File: modules/installation-and-upgrade/pages/container-deployment/suma/server-migration-suma.adoc [hunk 6]
```diff
@@ -26,20 +29,22 @@ Self trusted GPG keys are not migrated.
 GPG keys that are trusted in the RPM database only are not migrated.
 Thus synchronizing channels with [command]``spacewalk-repo-sync`` can fail.
 
-The administrator must migrate these keys manually from the 4.3 installation **after migration**:
+The administrator must migrate these keys manually from the 4.3 installation to the container host after the actual server migration.
 
-1. Copy the keys from the source server to the container host of the destination server.
-2. Add the keys to the container with [command]``mgradm gpg add ...``.
+1. Copy the keys from the 4.3 server to the container host of the new server.
+2. Later, add the keys to the container with [command]``mgradm gpg add ...``.
```
Comment (rjmateus) [inline modules/installation-and-upgrade/pages/container-deployment/suma/server-migration-suma.adoc:35]:
```suggestion
2. Later, add each key to the migrated server with the [command]``mgradm gpg add <PATH_TO_KEY_FILE>``.
```

**Review by rjmateus — APPROVED** [review]
Just one small comment. Other then that, looks good to me

**Review by jcayouette — APPROVED** [review]

**Pull request merged**